src/main/java/org/apache/sling/resourceresolver/impl/ResourceResolverFactoryActivator.java

```diff
@@ -304,6 +312,25 @@ protected void activate(final BundleContext bundleContext, final ResourceResolve
             this.observationPaths[i] = new Path(paths[i]);
         }
 
+        // optimize alias path allow list
+        this.aliasPathAllowList = null;
```

Comment:
I think you should swap/null out the field at the end, otherwise there is a window of opportunity when this field is null during recalculation.

Reply:
+1 - either compute the new data-structure and swap it out with a single assingment, or use a concurrent data-structure that could be updated incrementally. I would prefer the former, as it reflects the atomic nature of a config update better.

Reply:
As per my understanding, Null is set in the activate method before populating "aliasPathAllowList" to avoid any overlap. Since we are loading this variable using the configured paths, it can not be set null at the end

Reply:
Ah, I missed the fact that this was done in the `activate` method, so no one can read it at that point. But why do you need to explicitly set it to null? It already has that value since no one has changed it.

Reply:
I followed the initialization done for vanityPath related variables eg. vanityPathWhiteList
So initialized it with Null

Reply:
You're right @akankshajain18. However, not all committed code is free from code-smells. I think there's a fine line between emulating the style of an existing code-base, and copying bad practices without questioning them.